src/txn/txn_ckpt.c

```diff
@@ -1838,11 +1839,13 @@ __wt_checkpoint_close(WT_SESSION_IMPL *session, bool final)
     }
 
     /*
-     * Don't flush data from trees when there is a stable timestamp set: that can lead to files that
-     * are inconsistent on disk after a crash.
+     * Don't flush data from modified trees independent of system-wide checkpoint when either there
+     * is a stable timestamp set or the connection is configured to disallow such operation.
+     * Flushing trees can lead to files that are inconsistent on disk after a crash.
      */
-    if (btree->modified && !bulk && S2C(session)->txn_global.has_stable_timestamp &&
-      !__wt_btree_immediately_durable(session))
+    if (btree->modified && !bulk && !__wt_btree_immediately_durable(session) &&
+      (S2C(session)->txn_global.has_stable_timestamp ||
+          (!F_ISSET(S2C(session), WT_CONN_FILE_CLOSE_SYNC) && !metadata && !final)))
```

Comment:
In the description in `api_data.py` you say "If set to true... API calls that require exclusive access to tables will return EBUSY" yet here we are checking for 'false' to return EBUSY.

Reply:
Good catch - the documentation has wrong information. We return EBUSY when config is false.